test/create_env_cluster_topics_test.go

```diff
@@ -76,30 +57,189 @@ func TestEnvClusterTopic(t *testing.T) {
+	log.Printf("Actual ACL count: %d", aclCount)
+	return aclCount
+}
 
-		output = terraform.Output(t, runOptions, "environment_id")
-		a.NotEmpty(output)
+// Logic to validate individual ACL combinations (ResourceType and Operation) via Confluent CLI
+func validateIndividualKafkaACLDetails(t *testing.T, expectedACLs []struct {
+	ResourceType string
+	Operation    string
+	Count        int
+}) {
+
+	output := runCLICommand(t, "confluent", "kafka", "acl", "list", "-o", "json")
+	var acls []map[string]interface{}
+	err := json.Unmarshal([]byte(output), &acls)
+	assert.NoError(t, err, "Failed to parse Kafka ACL JSON output")
+
+	// Count ACLs by ResourceType and Operation
+	actualCounts := make(map[string]int)
+	for _, acl := range acls {
+		key := fmt.Sprintf("%s|%s", acl["resource_type"], acl["operation"])
+		actualCounts[key]++
+	}
 
-		output = terraform.Output(t, runOptions, "kafka_cluster_basic_name")
-		a.True(strings.Contains(output, "labs-kafka-cluster-"))
+	// Validate actual counts against expected values
+	for _, expected := range expectedACLs {
+		key := fmt.Sprintf("%s|%s", expected.ResourceType, expected.Operation)
+		actualCount := actualCounts[key]
+		assert.Equal(t, expected.Count, actualCount, fmt.Sprintf(
+			"Expected %d ACLs for ResourceType: %s, Operation: %s, but found %d",
+			expected.Count, expected.ResourceType, expected.Operation, actualCount,
+		))
+	}
 
-		output = terraform.Output(t, runOptions, "topic_service_account_key")
-		a.NotEmpty(output)
+	log.Println("Individual ACL combination validation successful.")
+}
 
-		output = terraform.Output(t, runOptions, "kafka_topic_name")
-		a.Equal(output, "squad_raw_service_example_1_entity")
+// Logic to validate Terraform outputs with Confluent CLI
+func validateTerraformOutputWithCLI(t *testing.T, envID, environmentName, kafkaClusterID, kafkaClusterBasicName, bigQueryConnectorID, gcsSinkConnectorID, kafkaTopicName1, kafkaTopicName2, topicServiceAccountID string) {
+	a := assert.New(t)
+
+	// Validate Environment ID and Name
+	log.Println("Validating Environment ID and Name...")
+	envOutput := runCLICommand(t, "confluent", "env", "describe", envID, "-o", "json")
+	var envDetails map[string]interface{}
+	a.NoError(json.Unmarshal([]byte(envOutput), &envDetails), "Failed to parse environment details")
+	a.Equal(envID, envDetails["id"], "Environment ID mismatch")
+	a.Equal(environmentName, envDetails["name"], "Environment Name mismatch")
+	log.Printf("Validated Environment ID '%s' and Name '%s' successfully.", envID, environmentName)
+
+	// Validate Kafka Cluster ID and Basic Name
+	log.Println("Validating Kafka Cluster ID and Basic Name...")
+	clusterOutput := runCLICommand(t, "confluent", "kafka", "cluster", "describe", kafkaClusterID, "-o", "json")
+	var clusterDetails map[string]interface{}
+	a.NoError(json.Unmarshal([]byte(clusterOutput), &clusterDetails), "Failed to parse Kafka Cluster details")
+	a.Equal(kafkaClusterID, clusterDetails["id"], "Kafka Cluster ID mismatch")
```

Comment:
These assertions seem ineffective - we should be validating our input (the values we specified in the Terraform example/module) with the actual infra created.

In this case we're just validating the Terraform output (which comes from the Confluent API anyways) with the value we get from the Confluent CLI. If we want to do this that's fine but we also need to validate against the input values.